packages/esm-patient-forms-app/src/forms/forms-see-all.component.tsx

```diff
@@ -0,0 +1,134 @@
+import Switch from 'carbon-components-react/es/components/Switch';
+import FormView from './form-view-see-all.component';
+import styles from './forms.component.scss';
+import { EmptyState, ErrorState } from '@openmrs/esm-patient-common-lib';
+import { useTranslation } from 'react-i18next';
+import { fetchAllForms, fetchPatientEncounters } from './forms.resource';
+import { filterAvailableAndCompletedForms } from './forms-utils';
+import { Encounter, Form } from '../types';
+import EmptyFormView from './empty-form.component';
+import { fhir } from '@openmrs/esm-framework';
+
+enum FormViewState {
+  recommended = 0,
+  completed,
+  all,
+}
+
+interface FormsProps {
+  patientUuid: string;
+  patient: fhir.Patient;
+}
+
+const Forms: React.FC<FormsProps> = ({ patientUuid, patient }) => {
+  async function setPatient(uUid) {
+    patient = (await fhir.read<fhir.Patient>({ type: 'Patient', patient: uUid })).data;
+  }
+
+  const query = new URLSearchParams(location.search);
+  patientUuid = query.get('patientUuid');
+  setPatient(patientUuid);
+  const { t } = useTranslation();
+  const displayText = t('forms', 'Forms');
+  const headerTitle = t('forms', 'Forms');
+  const [error, setError] = React.useState(null);
+  const [forms, setForms] = React.useState<Array<Form>>([]);
+  const [encounters, setEncounters] = React.useState<Array<Encounter>>([]);
+  const [completedForms, setCompletedForms] = React.useState<Array<Form>>([]);
+  const [selectedFormView, setSelectedFormView] = React.useState<FormViewState>(FormViewState.all);
+  const [filledForms, setFilledForms] = React.useState<Array<Form>>([]);
+
+  React.useEffect(() => {
+    fetchAllForms().subscribe((forms) => setForms(forms), setError);
+  }, []);
+
+  React.useEffect(() => {
+    const fromDate = dayjs(new Date()).startOf('day').subtract(500, 'day');
+    const toDate = dayjs(new Date()).endOf('day');
+    fetchPatientEncounters(patientUuid, fromDate.toDate(), toDate.toDate()).subscribe(
+      (encounters) => setEncounters(encounters),
+      setError,
+    );
+  }, [patientUuid]);
+
+  React.useEffect(() => {
+    const availableForms = filterAvailableAndCompletedForms(forms, encounters);
+    const completedForms = availableForms.completed.map((encounters) => {
+      encounters.form.complete = true;
+      encounters.form.lastCompleted = encounters.encounterDateTime ? encounters.encounterDateTime : null;
+      return encounters.form;
+    });
+    setCompletedForms(completedForms);
+  }, [forms, encounters]);
+
+  React.useEffect(() => {
+    const filledForms = forms.map((form) => {
+      completedForms.map((completeForm) => {
+        if (completeForm.uuid === form.uuid) {
+          form.complete = true;
+          form.lastCompleted = completeForm.lastCompleted ? completeForm.lastCompleted : null;
+        }
+      });
+      return form;
+    });
+    setFilledForms(filledForms);
+  }, [forms, completedForms]);
+
+  const RenderForm = () => {
```

Comment:
Hm not sure why we create a component in the component -> just render directly

Reply:
Will change this as well

Reply:
I have removed the extra component in the second commit